### PR TITLE
Add Fishing event hierarchy

### PR DESCRIPTION
**Pull request by Aaron1011**
This PR adds `Player` and `Human` versions of the following events (e.g `PlayerRetractFishingLineEvent`):

* `FishEvent`
 * `CastFishingLineEvent`
 * `HookedEntityEvent`
 * `RetractFishingLineEvent`

### CODE REVIEW

File: src/main/java/org/spongepowered/api/entity/projectile/FishHook.java [hunk 1]
```diff
@@ -35,21 +35,17 @@
...
-     * used to cast this fish hook. The hooked item may also be null.</p>
+     * Sets the hooked entity for this fish hook.
      *
-     * @param item The hooked item
+     * @param entity The hooked entity
      */
-    void setHookedItem(@Nullable ItemStack item);
+    void setHookedEntity(@Nullable Entity entity);
 
-}
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/entity/projectile/FishHook.java]: EOF is supposed to have a new line.

File: src/main/java/org/spongepowered/api/event/entity/living/player/fishing/PlayerRetractFishingLineEvent.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * a fishing line.
+ */
+public interface PlayerRetractFishingLineEvent extends HumanRetractFishingLineEvent, PlayerFishEvent {
+
+    /**
+     * Gets the {@link org.spongepowered.api.item.inventory.ItemStack} that will be given, if available.
+     *
+     * @return The {@link org.spongepowered.api.item.inventory.ItemStack} that will be given
+     */
+    Optional<ItemStack> getCaughtItem();
```
Comment (ST-DDT) [inline src/main/java/org/spongepowered/api/event/entity/living/player/fishing/PlayerRetractFishingLineEvent.java]: Why do you declare the methods twice?
They are inherited from the human version, arent they?

Comment (Aaron1011) [inline src/main/java/org/spongepowered/api/event/entity/living/player/fishing/PlayerRetractFishingLineEvent.java]: Oops, that was an accident. Thanks for finding that!

File: src/main/java/org/spongepowered/api/event/entity/living/human/fishing/package-info.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.entity.living.human.fishing;
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/human/fishing/package-info.java]: Newline missing

File: src/main/java/org/spongepowered/api/event/entity/living/player/fishing/package-info.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.entity.living.player.fishing;
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/player/fishing/package-info.java]: Same here.

File: src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanHookedEntityEvent.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     * Gets the {@link org.spongepowered.api.entity.Entity} hooked, if available.
+     *
+     * @return The hooked {@link org.spongepowered.api.entity.Entity}
+     */
+    Optional<Entity> getCaughtEntity();
+
+    /**
+     * Sets the {@link Entity} hooked, if available.
+     *
+     * @param entity The hooked {@Entity} to set
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanHookedEntityEvent.java]: `@Entity` is not a javadoc tag

File: src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanRetractFishingLineEvent.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+     * Gets the {@link Entity} hooked, if available.
+     *
+     * @return The hooked {@link Entity}
+     */
+    Optional<Entity> getCaughtEntity();
+
+    /**
+     * Sets the {@link Entity} hooked, if available.
+     *
+     * @param entity The hooked {@Entity} to set
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanRetractFishingLineEvent.java]: `@Entity` is not a javadoc tag

File: src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanFishEvent.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Called when a {@link org.spongepowered.api.entity.living.Human} performs
+ * a fishing-related action.
+ */
+public interface HumanFishEvent extends HumanEvent, Cancellable {
+
+    /**
+     * Gets the {@link org.spongepowered.api.entity.projectile.FishHook} used in this event
+     * @return
```
Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/human/fishing/HumanFishEvent.java]: No `@return` description.

File: src/main/java/org/spongepowered/api/event/entity/living/player/fishing/package-info.java (hunk 4 above)
Comment (Aaron1011) [inline src/main/java/org/spongepowered/api/event/entity/living/player/fishing/package-info.java]: There's actually no newline missing.

Comment (kenzierocks) [inline src/main/java/org/spongepowered/api/event/entity/living/player/fishing/package-info.java]: There was.....

Strange.

**Pull request closed without merging**